### PR TITLE
Add CI job to build PyVelox for Mac.

### PR DESCRIPTION
**Pull request by kgpai**
PyVelox requires miniconda so ensure that its installed from the setup-macos.sh script.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -196,7 +196,7 @@ jobs:
           # to manually avoid the case where bad dependencies
           # are cached, and has no other meaning.
           # If you update it, be sure to update save_cache too.
-          key: velox-circleci-macos-deps-v1-{{ checksum ".circleci/config.yml" }}-{{ checksum "scripts/setup-macos.sh" }}
```
Comment (kgpai) [inline .circleci/config.yml:199]: Forcing key change here so as to update the cache - which for some reason doesnt seem to get updated even if setup-macos.sh has changed.

**Comment by pedroerp** [comment]
Looks like the job failed.

**Comment by pedroerp** [comment]
Looks like the new job took 20 mins to run, making the full macos build take 1h30. Is this expected?

**Comment by kgpai** [comment]
@pedroerp The large amount of time in the macos build comes from the dependency cache being not cached - since in this PR we are adding an additional dependency. Once this is cached it should be faster. 
The pyvelox package build shouldnt take 20 minutes - I will work with @assignUser to see if we can speed it up.

**Comment by assignUser** [comment]
With #3884 merged and building pyvelox on linux and mac on each PR is this addition to CCI still useful?

**Comment by assignUser** [comment]
I think this is no longer needed and stale, feel free to reopen if needed @kgpai

**Pull request closed without merging**